scripts/update-browser-releases/edge.ts

```diff
@@ -0,0 +1,271 @@
+
+  //
+  // Get the edge.json from the local BCD
+  //
+  const file = fs.readFileSync(`${options.bcdFile}`);
+  const edgeBCD = JSON.parse(file.toString());
+
+  //
+  // Update the three channels
+  //
+  const channels = new Map([
+    ['current', options.releaseBranch],
+    ['beta', options.betaBranch],
+    ['nightly', options.nightlyBranch],
+  ]);
+
+  // Information store to be used outside the for loop
+  const data = {};
+
+  for (const [key, value] of channels) {
+    //
+    // Extract the useful data
+    //
+
+    // Find the right entry
+    const entry = edgeVersions.find((e) => e['Product'] === value);
+
+    // Get the last version
+    const version = parseFloat(entry['Releases'][0]['ProductVersion']);
+
+    // We need to find the first release with this version.
+    // As, they are from the newer to the older in the JSON object,
+    // we are looking for the first that is not the latest version
+    // and get the one immediately above (the first release of that
+    // version)
+    //
+    // The full version number is needed for the release note URL
+    // as well as the date
+    let id = 1;
+    while (
+      id < entry['Releases'].length &&
+      parseFloat(entry['Releases'][id]['ProductVersion']) === version
+    ) {
+      id++;
+    }
+    id--;
+
+    data[value] = {};
+    data[value].version = parseFloat(entry['Releases'][id]['ProductVersion']);
+    data[value].fullVersion = entry['Releases'][id]['ProductVersion'];
+
+    // NOTE: the published date is the one of the actual release
+    // (being the Stable, Beta, or Nightly),
+    // and not the one of the future release like we would like
+    // So we only get it if we are on the 'current' channel.
+
+    // Get published date
+    if (key === 'current') {
+      data[value].versionDate = entry['Releases'][id][
+        'PublishedTime'
+      ].substring(0, 10); // Remove the time part;
+    } else {
+      data[value].versionDate = await getFutureReleaseDate(
+        data[value].version,
+        options.releaseScheduleURL,
+      );
+    }
+
+    //
+    // Update the JSON in memory
+    //
+
+    // Get the release notes
+    const releaseNotesURL = await getReleaseNotesURL(
+      value,
+      data[value].fullVersion,
+      data[value].versionDate,
+    );
+
+    // Update in memory
+    if (
+      edgeBCD.browsers[options.bcdBrowserName].releases[data[value].version]
+    ) {
+      // The entry already exists
+      updateBrowserEntry(
+        edgeBCD.browsers[options.bcdBrowserName].releases[data[value].version],
+        data[value]?.versionDate,
+        key,
+        releaseNotesURL,
+      );
+    } else {
+      // New entry
+      newBrowserEntry(
+        edgeBCD,
+        options.bcdBrowserName,
+        data[value].version,
+        key,
+        options.browserEngine,
+        data[value]?.versionDate,
+        releaseNotesURL,
+      );
+    }
+  }
+
+  //
+  // Check that all older releases are 'retired'
```

Comment:
One more thing that we might want to do with older releases (but most probably fine to do in a follow-up PR): re-writing some of the release note URLs that moved from https://learn.microsoft.com/en-us/deployedge/microsoft-edge-relnote-stable-channel to https://learn.microsoft.com/en-us/deployedge/microsoft-edge-relnote-archive-stable-channel

Indeed, the stable channel release notes article only ever contains a certain number of releases. To avoid making this article very long, releases that are older than some amount of time are moved to another article.

If you look at https://github.com/mdn/browser-compat-data/blob/main/browsers/edge.json, you'll see that the current cut off is before 115 (as in, 115 is in the main release notes article, but 114 is in the archived release notes article).

In the semi-automated workflow I used to use, I had a playwright-based script that would read both pages and find the corresponding `<h2>` section headings to know where a release note for a given release was.

Reply:
This is an important point. I wrote it down to do it in a follow-up.